lib/colorado_lottery.rb

```diff
@@ -0,0 +1,40 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants,
+              :name
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game_name)
+    if contestant.game_interests != game_name.name
+      return false
+    elsif contestant.game_interests == game_name.name && contestant.age >= 18
+      true
+    end
+  end
+
+  def can_register?(contestant, game_name)
+    if contestant.age >= 18 || contestant.game_interests == game_name.name
+      return true
+    end
+  end
+
+  def register_contestant(contestant, game_name)
+    @registered_contestants[game_name] = contestant
+  end
+
+  def eligible_contestants(game_name)
+    eligible_contestants = []
+    @registered_contestants.each do |contestant|
+      eligible_contestants << contestant
```

Comment:
right idea, but a couple things to remember: 

1) `@registered_contestants` is a hash, so each individual element is not a single contestant-- it would be a key/value pair, the way you have it written here. 

2) you only wanted to grab the contestant _if_ they had enough spending money to afford the game, so you'd need a conditional inside of this each. 

```
def eligible_contestants(game_name)
    eligible_contestants = []
    @registered_contestants.each do |game, contestant|
     if contestant.spending_money >= game_name.cost
         eligible_contestants << contestant
     end 
     eligible_contestants.uniq
end
```

That would work okay, but isn't ideal, because you're iterating over every contestant for every game, when the method really only cares about the one game being passed in. So this would be a better solution: 

```
  def eligible_contestants(game)
    registered_contestants[game.name].find_all {|contestant| contestant.spending_money >= game.cost}
  end
```